### ISSUE TITLE
[Thesis need] What's the best way to get data?

### ISSUE THREAD

**Issue by gymgym1212** [X-lab2017/open-digger#872]
Hi community, nowadays I have recently been thinking about how to get the data I need for my thesis experiment.
Now that I have a list of CII repositories, I want to get all the repositories and developers within 6 hops of the GitHub-wide collaboration network from those repositories and use that data to build my heterogeneous network of developer collaboration.

This is not a simple task, as it involves a lot of data and can be very time-consuming. So I wanted to ask what are the best practices for getting this done?  Notebook？ @frank-zsy @wengzhenjie @bifenglin

[cii_repo_list.csv](https://github.com/X-lab2017/open-digger/files/8897446/cii_repo_list.csv)

**Comment by frank-zsy** [comment]
First you need to define how the collaboration network build, what the node and edge is. And 6 hops may contain lots of data like almost all the network. And you may need to constrain the time range, when the time range expands, the data will increase too. And maybe I can do this for you on the graph database.

**Comment by gymgym1212** [comment]
Many thanks! @frank-zsy 
I'm still considering the exact structure of the diagram, and I'll continue to leave comments as I determine it. Then I will learn your data processing flow, because I may adjust the graph structure and can't always bother you.

**Comment by frank-zsy** [comment]
@gymgym1212 Here is some data I tried with the heterogeneous collaborate network with activity as edge and user/repo as node. Only starting from CII badge repo with no time range limit, nodes returned for 2 hops are 67,323, for 3 hops are 318,0757 and for 4 hops are 1,143,5225. 

And if I tried to get data for 6 hops, the query can not return in 10 hours. And I guess if you start with more than 4,000 nodes, the nodes returned in 4 hops can be enormous like maybe 10 million nodes which is 1/3 of the whole network and the whole network may return in 6 hops.
